File: Clases_web/MundoPC/src/cl/com/gm/mundopc/Orden.java
package cl.com.gm.mundopc;

public class Orden {
    private final int idOrden;
    private Computadora computadoras[];
    private static int contadorOrdenes;
    //saber el max de pc que podemos agregar al arreglo
    private static final int MAX_COMPUTADORAS = 10;
    //saber cuantos objetos se han agregado al arreglo
    //asi evitamos la iteracion todo los que no son null
    private int contadorComputadoras;
    
    public Orden(){
        this.idOrden = ++Orden.contadorOrdenes;
        this.computadoras = new Computadora[Orden.MAX_COMPUTADORAS];
    }
    
    public void agregarComputadora(Computadora computadora){
        if(this.contadorComputadoras < Orden.MAX_COMPUTADORAS){
            this.computadoras[this.contadorComputadoras++] = computadora;
        }
        else{
            System.out.println("Has superado el limite" + Orden.MAX_COMPUTADORAS);
        }
    }
    
    public void motrarOrden(){
        System.out.println("Orden #:" + this.idOrden);
        System.out.println("Computadoras de la orden #:" + this.idOrden);
        for (int i = 0; i < this.contadorComputadoras; i++) {
            System.out.println(this.computadoras[i]);
        }
    }
}

File: JavaScript/SistemaComputadora/Orden.js
class Orden {
    static contadorOrdenes = 0;

    constructor(){
        this._idOrden = ++Orden.contadorOrdenes;
        this._computadoras = [];
    }
    get idOrden(){
        return this._idOrden;
    }
    agregarComputadora(computadora){
        this._computadoras.push(computadora)
    }

    mostrarOrden(){
        let computadorasOrden = ' ';
        for(let computadora of this._computadoras){
            computadorasOrden += `\n${computadora}`;
        }
        console.log(`Orden: ${this._idOrden}, computadoras: ${this.computadorasOrden}`)
    }
}

let orden1 = new Orden();
orden1.agregarComputadora(computador1);
orden1.agregarComputadora(computador1);
orden1.agregarComputadora(computador1);
orden1.mostrarOrden();